eclipse-collections-code-generator/src/main/resources/impl/list/mutable/primitiveArrayList.stg

```diff
@@ -1003,6 +1004,25 @@ public class <name>ArrayList extends Abstract<name>Iterable
         return this;
     }
 
+    /**
+     * Randomly permutes this list in place and returns the same list (this).
+     *
+     * Implements the Fisher-Yates shuffle algorithm using {@code java.util.Random} as the source of randomness.
+     */
+    @Override
+    public <name>ArrayList shuffleThis()
```

Comment:
Why not have a `shuffleThis(Random)` and have this method call that one? Both methods should have default method implementations though so this can be introduced safely in a minor release.